src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -1075,6 +1070,60 @@ public void ValidateRelationshipsInConfig(RuntimeConfig runtimeConfig, IMetadata
         }
     }
 
+    /// <summary>
+    /// Handles the validation of source and target fields as valid backing columns in the DB.
+    /// </summary>
+    /// <param name="relationshipName">Name of the relationship.</param>
+    /// <param name="entityName">The name of the entity that we check for backing columns.</param>
+    /// <param name="relationship">The relationship that holds the fields to validate.</param>
+    /// <param name="sqlMetadataProvider">The sqlMetadataProvider used to lookup if the backing fields are valid columns in DB.</param>
+    /// <param name="invalidColumns">List in which invalid fields are aggregated, this list may be modified by this function.</param>
+    private void ValidateSourceAndTargetFieldsAsBackingColumns(
+        string entityName,
+        string relationshipName,
+        EntityRelationship relationship,
+        ISqlMetadataProvider sqlMetadataProvider,
+        List<string> invalidColumns)
+    {
+        // In all kinds of relationships, if sourceFields are included they must be valid columns in the backend.
+        if (relationship.SourceFields is not null)
+        {
+            GetFieldsNotBackedByColumnsInDB(
```

Comment:
As a side effect of this method, you are populating the invalidColumns list which is not evident by the function name `GetFieldsNotBackedByColumnsInDB` which suggests getting back list of invalid fields. I think returning a list from this method call would be a better option.

Reply:
I should also mention, this code is from a previous PR, but it has been moved as part of this PR.